src/components/FileUploadModal/index.tsx

```diff
@@ -0,0 +1,117 @@
+import { Button, Form, Tabs } from "antd";
+import { RcFile } from "antd/lib/upload";
+import React, { useState } from "react";
+import { ActionCreator } from "redux";
+
+import {
+    ClearSimFileDataAction,
+    RequestLocalFileAction,
+} from "../../state/trajectory/types";
+import {
+    SetErrorAction,
+    SetViewerStatusAction,
+} from "../../state/viewer/types";
+
+import CustomModal from "../CustomModal";
+import UrlUploadForm from "./url-upload-form";
+import LocalFileUpload from "../LocalFileUpload";
+import uploadFiles from "./upload-local-files";
+
+import styles from "./style.css";
+
+interface FileUploadModalProps {
+    setIsModalVisible: (isModalVisible: boolean) => void;
+    loadLocalFile: ActionCreator<RequestLocalFileAction>;
+    setViewerStatus: ActionCreator<SetViewerStatusAction>;
+    clearSimulariumFile: ActionCreator<ClearSimFileDataAction>;
+    setError: ActionCreator<SetErrorAction>;
+}
+
+const FileUploadModal: React.FC<FileUploadModalProps> = ({
+    setIsModalVisible,
+    clearSimulariumFile,
+    loadLocalFile,
+    setViewerStatus,
+    setError,
+}) => {
+    const [openTab, setOpenTab] = useState("device");
+    const [noUrlInput, setNoUrlInput] = useState(true);
+    const [fileList, setFileList] = useState<RcFile[]>([]);
+    const [urlForm] = Form.useForm();
```

Comment:
I haven't used this before, it seems neat

Reply:
I also just learned about this. Seems worth keeping in mind for the rest of autoconversion work, which may require a similar pattern of splitting a single logical form across multiple child components.